lib/components/Masthead/Masthead.tsx

```diff
@@ -4,167 +4,85 @@ import { MethodNode } from '../../Common';
 import * as InlinePopup from '../InlinePopup';
 import { Thumbnail } from '../Thumbnail';
 import { NavigationProperties } from '../Navigation/Navigation';
-import { ActionTriggerButton } from '../ActionTrigger';
-import { ContentPanel, ContentPanelProperties } from '../ContentPanel';
+import { ActionTriggerButton, ActionTriggerAttributes, ActionTriggerButtonAttributes } from '../ActionTrigger';
 import { Elements as Attr } from '../../Attributes';
+import { SearchInput } from '../SearchInput/SearchInput';
+import { TextInputAttributes } from '../Input/TextInput';
 
 const cx = classnames.bind(require('./Masthead.module.scss'));
 
-export interface MastheadAttributes {
-    userMenuAriaLabel?: string;
+interface ToolbarItem {
+    icon: string; // TODO: update with MethodNode
+    label: string;
+    isSelected: boolean;
+    onClick: (event: any) => void;
+    attr?: ActionTriggerButtonAttributes & ActionTriggerAttributes;
 }
 
-interface ToolbarItem {
-    icon: string;
-    contentPanel?: ContentPanelProperties;
-    className?: string;
-    ariaLabel?: string;
+interface SearchItem {
+    label: string;
+    value: string;
+    hidden?: boolean;
+    onSubmit: (event: any) => void;
+    onChange: (event: any) => void;
+    onIconClick?: (event: any) => void;
+    attr?: TextInputAttributes;
+}
+
+interface UserItem {
+    userMenuAriaLabel?: string;
+    onUserMenuClick?: (event: any) => void;
+    userMenuExpanded?: boolean;
+    userMenuItems?: MethodNode;
 }
 
 export interface MastheadProperties {
     branding: MethodNode;
     navigation?: NavigationProperties;
-    toolBarItems: {
-        search?: {
-            key: string;
-            label: string;
-            onClick: Function;
-        };
-        settings: ContentPanelProperties;
-        help: ContentPanelProperties;
-    };
-    userItemAttr?: MastheadAttributes;
+    search?: SearchItem;
+    more: ToolbarItem;

```

Comment:
should more and toolbarItems be optional as well? We might need to remove it for the generic 4xx/5xx pages

Reply:
good point. added as optional